frontend/src/pages/CommentForm.js

```diff
@@ -4,62 +4,55 @@ import { useComment } from "../hooks/useComment";
 import { useAuthContext } from "../hooks/useAuthContext";
 
 const CommentForm = (props) => {
-  const { dispatch } = useComment();
-  const { user } = useAuthContext();
-  const [isSubmitted, setShowSubmittedMessage] = useState(false);
-
-
-  const id = user.id;
-
-  const [userName, setUserName] = useState("");
-  const { addComment, error, loading } = useComment();
-
-  useEffect(() => {
-    getUserName();
-  }, []);
-
-  const getUserName = async () => {
-    const response = await fetch(`/api/org-users/${id}`);
-    const json = await response.json();
-    if (response.ok) {
-      if (json.organisationName) {
-        setUserName(json.organisationName);
-      } else {
-        const response = await fetch(`/api/ind-users/${id}`);
-        const json = await response.json();
-        if (response.ok) {
-          setUserName(json.firstName);
-        }
-      }
-    } else {
-      dispatch({ type: "SET_ERROR", payload: json });
-    }
-  };
-
-  const handleSubmit = async (e) => {
-    e.preventDefault();
-    const { content } = e.target.elements;
-
-    await addComment(userName, content.value);
-    setShowSubmittedMessage(true)
-  };
-
-  return (
-    <main>
-      <div className="container">
-        <form className="addCommentForm" onSubmit={handleSubmit}>
-          <div className="input-box">
-            <input className="form_field" placeholder="Add a comment..." id="content" type="text"/>
-            <i></i>
-          </div>
-
-          <input id="submit" type="submit" value="Post" disabled={loading} />
-        </form>
-        {error && <div className="error">{error}</div>}
-        {isSubmitted && <div className="info">Comment Submitted!</div>}
-      </div>
-    </main>
-  );
+    const { dispatch } = useComment();
+    const { user } = useAuthContext();
+    const { addComment, error, loading } = useComment();
+
+    const [isSubmitted, setShowSubmittedMessage] = useState(false);
+    const [userName, setUserName] = useState("");
+
+    const id = user.id;
+    const userType = user.type;
+
+    useEffect(() => {
+        const getUserName = async () => {
+            const response = await fetch(`/api/${userType}-users/${id}`);
+            const json = await response.json();
+            if (response.ok) {
+                if (userType === 'org') setUserName(json.organisationName);
+                if (userType === 'ind') setUserName(json.firstName + " " + json.lastName);
+            } else {
+                dispatch({ type: "SET_ERROR", payload: json });
+            }
+        };
+
+        getUserName();
+    }, [userName]);
```

Comment:
bit of trial and error and found this was a much simpler way of setting the username and fetching with a second template literal

Reply:
yeah in hindsight maybe user name would have made more sense at the start of the project, my bad! we'll implement it in the next iteration ;)